### ISSUE TITLE
Add KaTeX rendering for Org-mode files

### ISSUE THREAD

**Issue by tecosaur** [go-gitea/gitea#21630]
### Feature Description

Now that [KaTeX rendering for Markdown](https://github.com/go-gitea/gitea/pull/20571) has been merged, it would be nice to extend that to Org documents too.

I've made a quick test/about document on try.gitea: https://try.gitea.io/tecosaur/testing/src/branch/main/gitea-maths-test.org

The short version is that it seems like go-org (which gitea currently uses) already identifies maths, but nothing is done with it. I would recommend targeting the following constructs:
- `\( inline maths \)`
- `\[ display maths \]`
- `\begin{env} latex env \env{env}`

**Comment by eeyrjmr** [comment]
I guess the proposal should be more for Katex to work in md files not just comments.

**Comment by delvh** [comment]
I think that is exactly what this issue is trying to say, but you need the knowledge what go-org is…

**Comment by tecosaur** [comment]
I'm not sure about the support for LaTeX constructs in other plaintext markup formats Gitea supports, I just know that go-org (https://github.com/niklasfasching/go-org) is used for Org (`.org` files), and it recognises them.

**Comment by holocronweaver** [comment]
FWIW for org-mode latex I typically use dollar signs for inline (`$`) and display (`$$`), though [the single-dollar has some slightly wonky rules for recognition](https://orgmode.org/manual/LaTeX-fragments.html).

Would be happy to see any support though, especially for `\(` and `\[`.

**Comment by tecosaur** [comment]
Yea, `$` are soon to be recommended against in the official manual, there's even been talk of removing them from the syntax on the mailing list (not going to happen due to backwards compatibility concerns). So, I'd consider supporting `\(` and `\[` much more important than `$`/`$$`.

**Comment by holocronweaver** [comment]
> `$` are soon to be recommended against in the official manual

That's unfortunate, though I can understand why, parsing would be wonky. Still, in my limited experience `$` is fairly common in the wild, and as you said it isn't going away anytime soon, so would be nice to have it for compatibility. But I agree `\(` and `\[` should have higher priority, though wide compatibility is pretty important too so having `$` support would still be great.

**Comment by JirR02** [comment]
Is it possible to solve this issue by adding [MathJax](https://www.mathjax.org/#gettingstarted) as a script or do you intend to implement this using go?

**Comment by JirR02** [comment]
> It works, you only need to render a "math" language code block:
> 
> ![Image](https://github.com/user-attachments/assets/91adcbb1-044f-49ee-9fea-84b43f3a0bf6)
> 
> ![Image](https://github.com/user-attachments/assets/4a54d576-1344-4d9e-89f9-b54f0d7d0825)
> 
> However, the orgmode https://orgmode.org/manual/LaTeX-fragments.html isn't supported yet

So I am guessing that they want to implement it using go?

**Comment by wxiaoguang** [comment]
> So I am guessing that they want to implement it using go?

Yes, only need to make the orgmode render module output HTML like `<code class="language-math"></code>`, then frontend KaTeX module will render it correctly.
